moonstone/analysis/statistical_test.py

```diff
@@ -0,0 +1,167 @@
+import numpy as np
+import pandas as pd
+from scipy import stats
+from statsmodels.stats.multitest import multipletests
+from typing import List, Union
+
+from moonstone.utils.pandas.series import SeriesBinning
+
+import logging
+logger = logging.getLogger(__name__)
+
+
+def statistical_test_groups_comparison(series: pd.Series, group_series: pd.Series, stat_test: str,
```

Comment:
The function itself is a bit too long and I think you could easily refactor it to increase the readibility of the code and facilitate testing too. Have a quick look to see wether you can easily refactor otherwise we can leave it like this.